.github/instructions/mastg-test.instructions.md

```diff
@@ -0,0 +1,208 @@
+## Tests
+
+Tests are platform-specific and must be located under tests-beta/android/ or tests-beta/ios/, within the corresponding MASVS category. Their file names are the test IDs.
+
+Notes:
+
+- Tests with `platform: network` are still organized under the OS folder that the MASVS category belongs to (for example, Android network tests live under `tests-beta/android/MASVS-NETWORK/`).
+
+Tests have an overview and contain Steps which produce outputs called observations. After following the Steps you come up with an [Observation](https://docs.google.com/document/d/1EMsVdfrDBAu0gmjWAUEs60q-fWaOmDB5oecY9d9pOlg/edit?pli=1#heading=h.h9gqgz4hdubj) which you will [Evaluate](https://docs.google.com/document/d/1EMsVdfrDBAu0gmjWAUEs60q-fWaOmDB5oecY9d9pOlg/edit?pli=1#heading=h.lare0v58fwbf).
+
+Example:
+
+```sh
+% ls -1 -F tests-beta/android/MASVS-CRYPTO/
+MASTG-TEST-0204.md
+MASTG-TEST-0205.md
+```
+
+### Markdown: Metadata
+
+#### title
+
+Test titles should be concise and clearly state the purpose of the test.
+
+In some cases, the test name and the weakness may have the same title, but typically tests cover different aspects of a weakness. Titles should reflect that.
+
+Avoid including Android or iOS unless necessary, as in "Insecure use of the Android Protected Confirmation API".
+
+Follow a consistent style across all test titles.
+
+**Conventions**
+
+- Static: “References to…” (semgrep/r2)
+- Dynamic: “Runtime Use …” (frida)
+
+Exceptions may apply where "Runtime ..." feels forced, for example tests using adb, local backups, or filesystem snapshots.
+
+Examples:
+
+- [MASTG-TEST-0207](https://mas.owasp.org/MASTG/tests-beta/android/MASVS-STORAGE/MASTG-TEST-0207/)
+- [MASTG-TEST-0216](https://mas.owasp.org/MASTG/tests-beta/android/MASVS-STORAGE/MASTG-TEST-0216/)
+- [MASTG-TEST-0263](https://mas.owasp.org/MASTG/tests-beta/android/MASVS-STORAGE/MASTG-TEST-0263/)
+
+#### platform
+
+The mobile platform. One of: ios, android, network.
+
+- Use network for platform-agnostic traffic analysis tests where the checks are performed purely on captured/observed traffic (often paired with type: [network]).
+
+#### id
+
+The test ID.
+
+#### weakness
+
+The MASWE weakness ID the test references.
+
+- In YAML front matter, specify the bare identifier (for example, `weakness: MASWE-0069`). In body text, include the leading `@` (for example, @MASWE-0069).
+
+#### type
+
+One or more test types.
+
+Supported: static, dynamic, network, manual, filesystem, developer.
+
```

Comment:
Static & dynamic are self-explanatory but could be defined. The rest are not very obvious, so a small description might help the reader. Sample wording:

```suggestion

static: performed without executing the application (e.g by reading the source code)
dynamic: performed by executing the binary and testing it within a (closed) runtime environment
network: TODO
manual: tests that you have to use the app
filesystem: TODO
```